mesonbuild/scripts/env2mfile.py

```diff
@@ -180,7 +185,8 @@ def detect_cross_debianlike(options: T.Any) -> MachineInfo:
     host_arch = data['DEB_HOST_GNU_TYPE']
     host_os = data['DEB_HOST_ARCH_OS']
     host_subsystem = host_os
```

Comment:
Is this right? In practice `DEB_HOST_ARCH_OS` takes values `linux`, `kfreebsd` and `hurd`.

The only subsystems that Meson currently documents are the various flavours of Darwin (`macos`, `ios`, etc.), and if anything I would have expected the closest equivalent of that on Linux to be the libc implementation (the "GNU" in GNU/Linux). In Debian that's `DEB_HOST_ARCH_LIBC`, which in practice is `gnu` on all extant Debian architectures and ports, but ports that use `musl` or `uclibc` have been proposed.

Reply:
Or possibly the Meson subsystem is more like the `DEB_BUILD_ARCH_ABI`, which is normally `base`, but can be for example `eabihf` on ARM EABI hardfloat.

Or, if we don't actually know what "subsystem" would really mean on non-Darwin, maybe it would be better to not include it in `env2mfile` output at all.

Reply:
I don't think we should set them except on MacOS, since they're not documented in the table, or we should document them and hav env2mfile massage values to our expected ones. And the methods aren't documented in the function reference. Sweet.

tl;dr This was all a sort of "best we can agree on" with subsystem. I really wanted to have a clear way to differentiate Android/Linux, GNU/Linux, Debian/FreeBSD, FreeBSD/FreeBSD, GNU/OpenSolaris, etc. We just couldn't really get an agreement on what to call those things.

Reply:
I'm going to turn this thread into a separate issue report.

Reply:
#13743